apps/entity-service/src/entities/processors/entity-processor.ts

```diff
@@ -87,12 +86,31 @@ export abstract class EntityProcessor<
 
   async getLightDtos(models: ModelType[]): Promise<DtoResult<LightDto>[]> {
     const results: DtoResult<LightDto>[] = [];
+
+    let associateData: Record<number, LightDto> = {};
+
+    const allowedTypes = [Project];
+    if (allowedTypes.some(type => models[0] instanceof type)) {
+      associateData = (await this.loadAssociationData(models.map(m => String(m.id)))) as Record<number, LightDto>;
+    }
+
     for (const model of models) {
-      results.push(await this.getLightDto(model));
+      let dto = await this.getLightDto(model);
+      dto = this.mergeDto(dto, associateData, model);
```

Comment:
Instead of this merge DTO pattern, the this method should just pass the associatedData property to getLightDto, and let the subclass handle it.